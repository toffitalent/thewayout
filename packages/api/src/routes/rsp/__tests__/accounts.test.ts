import { faker } from '@faker-js/faker';
import { auth, fixtures, request, resetDb } from '@test';

const { rsp1, rspAccount2, user7, user8 } = fixtures;

describe('Routes > Rsp > Accounts', () => {
  beforeEach(async () => {
    await resetDb();

    faker.seed(1);
  });

  describe('PUT /rsp/:id/accounts/:userId', () => {
    test('updates rsp member account', async () => {
      const res = await request()
        .put(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .use(auth(user8.id, ['rsp']))
        .send({ phone: '4142133945' });

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('only updates account by selfs', async () => {
      const res = await request()
        .put(`/api/v1/rsp/${rsp1.id}/${rspAccount2.userId}/accounts`)
        .use(auth(user7.id, ['rsp']))
        .send({ phone: '4142133945' });

      expect(res.status).toBe(404);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('GET /rsp/:id/accounts/case-managers', () => {
    test('list case manager names', async () => {
      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/accounts/case-managers`)
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });
});
