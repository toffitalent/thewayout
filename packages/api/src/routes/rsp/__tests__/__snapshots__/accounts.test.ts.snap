// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes > Rsp > Accounts GET /rsp/:id/accounts/case-managers list case manager names 1`] = `
[
  {
    "firstName": "Quinten",
    "id": "01817e24-0d42-7846-9a0b-4a6f781f0513",
    "lastName": "McDermott",
  },
  {
    "firstName": "Mariah",
    "id": "01817e24-0d42-786a-a454-7291c0173657",
    "lastName": "Barton",
  },
]
`;

exports[`Routes > Rsp > Accounts PUT /rsp/:id/accounts/:userId only updates account by selfs 1`] = `
{
  "code": "ResourceNotFound",
  "message": "Resource /api/v1/rsp/01817e24-0d42-77e5-9a94-ad514d8ec5dc/01817e24-0d42-75e5-8c38-8eb33c91884c/accounts not found",
  "status": 404,
}
`;

exports[`Routes > Rsp > Accounts PUT /rsp/:id/accounts/:userId updates rsp member account 1`] = `
{
  "avatar": null,
  "email": "mariah_barton@example.com",
  "firstName": "Mariah",
  "id": "01817e24-0d42-786a-a454-7291c0173657",
  "isProfileFilled": true,
  "lastName": "Barton",
  "phone": "4142133945",
  "role": "member",
  "rsp": {
    "address": "Ibrahim Pine",
    "avatar": null,
    "city": "East Barneybury",
    "createdAt": "<Date>",
    "description": "Enim sequi iusto.
Explicabo quisquam omnis nisi corporis corrupti.
Cupiditate qui culpa numquam iure quisquam vero architecto.",
    "email": null,
    "id": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
    "justiceStatus": [
      "halfwayHouse",
      "parole",
    ],
    "name": "Oberbrunner and Sons",
    "offenses": [
      "burglary",
      "distribution",
      "diuDwi",
      "embezzlement",
    ],
    "phone": null,
    "postalCode": "78580",
    "servicesArea": [
      "Adams",
      "Ashland",
      "Barron",
    ],
    "state": "AK",
    "support": [
      "basicHygieneSupport",
      "clothing",
      "basicTechSkills",
      "childCare",
    ],
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
    ],
  },
  "rspId": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
  "userId": "01817e24-0d42-75e5-8c38-8eb33c91884c",
}
`;
